### ISSUE TITLE
Feature Request: Compiler Option for Worker Scripts

### ISSUE THREAD

**Issue by si-robertson** [google/closure-compiler#1032]
Hi guys,

I apologise if this has already been discussed somewhere, I did search through the issues here but failed to find anything related to this request.

Would it be possible to add a compiler option that tells the compiler what `self` is? I could be doing something wrong, but it appears as though the compiler assumes `self` is always a `Window`.

Something as simple as `--worker` should be fine, but that's your call. Internally, that would provide a conditional switch for the default externs, i.e. either load the default window or worker externs. I may be able to add this feature myself if needed.

**Comment by dimvar** [comment]
see issue #966

**Comment by igrep** [comment]
Let me ask one.
Finally, how do we tell the environment where the compiled script runs to the compiler?
I checked both https://github.com/google/closure-compiler/pull/1047 and https://github.com/google/closure-compiler/issues/966 , but didn't get sure.

I tried to compile a ServiceWorker script with no `--externs` and no `--env` options,
but it complains:

```
some_module_in_worker.js:123: WARNING - Property registration never defined on Window
      self.registration.showNotification(title, options);
```

Or, is this feature still in progress?
